src/main/java/core/basesyntax/service/operations/SupplyOperation.java

```diff
@@ -0,0 +1,11 @@
+package core.basesyntax.service.operations;
+
+import core.basesyntax.model.FruitTransaction;
+
+public class SupplyOperation implements OperationHandler {
+    @Override
+    public void handle(FruitTransaction transaction) {
+        storageFruits.put(transaction.getFruit(), storageFruits
+                .getOrDefault(transaction.getFruit(), 0) + transaction.getQuantity());
+    }
+}
```

Comment:
The `SupplyOperation` class should not directly manipulate the storage. Instead, it should use a service to interact with the storage. This ensures that the class adheres to the SOLID principles, specifically the Single Responsibility Principle and the Dependency Inversion Principle. Consider injecting a storage service that provides methods to manipulate the storage.